pkg/controller.v1alpha1/xgboost/controller.go

```diff
@@ -39,6 +40,25 @@ import (
 	"time"
 )
 
+const (
+	controllerName = "xgboost-operator"
+
+	// labels for pods and servers.
+	replicaTypeLabel    = "xgboost-replica-type"
+	replicaIndexLabel   = "xgboost-replica-index"
+	// labelGroupName      = "group-name"
+	// labelXGBoostJobName = "xgboost-job-name"
+	labelXGBoostJobRole = "xgboost-job-role"
+)
+
+var (
+	// KeyFunc is the short name to DeletionHandlingMetaNamespaceKeyFunc.
+	// IndexerInformer uses a delta queue, therefore for deletes we have to use this
+	// key function but it should be just fine for non delete events.
+	KeyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
```

Comment:
Reuse this from common: https://github.com/kubeflow/common/blob/48d498ab85e1ff26b75c2a3023d77b465ca25ca8/job_controller/job_controller.go#L34